app/Http/Controllers/UserController.php

```diff
@@ -2,9 +2,83 @@
 
 namespace App\Http\Controllers;
 
-class UserController
+use App\Http\Resources\Authentication\UserResource;
+use Exception;
+use App\Models\User;
+use Illuminate\Http\Request;
+use Illuminate\Support\Facades\URL;
+use Illuminate\Support\Facades\Hash;
+use Illuminate\Support\Facades\Http;
+use App\Http\Resources\ErrorResource;
+use App\Http\Resources\SuccessResource;
+use Illuminate\Support\Facades\Validator;
+
+class UserController extends Controller
 {
 
+    public function clientToken(Request $request)
+    {
+        $validator = Validator::make($request->all(), [
+            'client_id' => 'required',
+            'client_secret' => 'required',
+        ]);
+
+        if ($validator->fails()) {
+            $res = (object)[
+                "status_code" => 400,
+                "message"     => "The given data was invalid.",
+                "error"       => $validator->errors(),
+                "data"        => null
+            ];
+            return (new ErrorResource($res))->response()->setStatusCode(400);
+        }
+        
+        try {
+            $response = Http::asForm()->post(URL::to('/') . '/oauth/token', [
+                'grant_type' => 'client_credentials',
+                'client_id' => $request->client_id,
+                'client_secret' => $request->client_secret,
+                'scope' => '*',
+            ]);
+    
+            $status = $response->status();
+            
+            if ($status === 200) {
```

Comment:
@kshitij1326 We might move it a separate PHP package. We need to avoid nested if else.